### ISSUE TITLE
putFileContents 401 Unauthorized while getDirectoryContents works

### ISSUE THREAD

**Issue by mnewmedia** [perry-mitchell/webdav-client#20]
```js
// works
client.getDirectoryContents("/").then(function(contents) {
   console.log(JSON.stringify(contents, undefined, 4));
}).catch(function(err) {
    console.error(err);
});

// fails: 401 Unauthorized
let fileContent = fs.readFileSync(newFilePath);
client.putFileContents("/myImage.jpg", fileContent, {format: "binary", "Content-Type": "application/octet-stream" }).catch(function(err) {
    console.error(err);
});
```
Error:
```js
Error: Invalid response: 401 Unauthorized
    at handleResponseCode (C:\projekte\ai\electron-ner\node_modules\webdav\source\adapter\response.js:6:19)
    at process._tickCallback (internal/process/next_tick.js:103:7)
````

Like this i could upload files:
```js
let request = require('request');
fs.createReadStream(newFilePath).pipe(request({
    method: 'PUT',
    url: 'https://domain/dav/file.jpg',
    auth: auth
}, function (error, response, body) {
    console.log('error', error);
    console.log('response', response);
    console.log('body', body);
}));
```

**Comment by perry-mitchell** [comment]
Hi @mnewmedia, thanks for finding this! It is indeed a bug and I'm about to patch it now..

**Comment by perry-mitchell** [comment]
Fixed in 0.4.1
